File: adedonhaJava/src/conjunto/Jogar.java
package conjunto;

public class Jogar{
    String me;
    String opponent;
    public Jogar(String myPlay, String opponentPlay){
        this.me = myPlay;
        this.opponent = opponentPlay;
    }

    public String compara(){
        if(this.me.equals("pedra")&&(this.opponent.equals("lagarto") || this.opponent.equals("tesoura"))){
            return "Vitoria";
        }
        else if(this.me.equals("spock")&&(this.opponent.equals("tesoura") || this.opponent.equals("pedra"))){
            return "Vitoria";
        }
        else if(this.me.equals("papel")&&(this.opponent.equals("pedra") || this.opponent.equals("spock"))){
            return "Vitoria";
        }
        else if(this.me.equals("lagarto")&&(this.opponent.equals("spock") || this.opponent.equals("papel"))){
            return "Vitoria";
        }
        else if(this.me.equals("tesoura")&&(this.opponent.equals("lagarto") || this.opponent.equals("papel"))){
            return "Vitoria";
        }
        else if(this.me.equals(this.opponent)){
            return "Empate";
        }
        else return "Derrota";
    }
}

File: adedonhaPython/jogar.py
from bancoDados import DataBase

class Competicao:
    def __init__(self,myPlay,opponentPlay):
        self.me = myPlay
        self.opponent = opponentPlay

    def compara(self):
        if self.me == 'pedra' and (self.opponent == 'lagarto' or self.opponent == 'tesoura') :
            return 'Vitoria'

        elif self.me == 'spock' and (self.opponent == 'tesoura' or self.opponent == 'pedra'):  
            return 'Vitoria' 

        elif self.me == 'papel' and (self.opponent == 'pedra' or self.opponent == 'spock'):  
            return 'Vitoria'
        
        elif self.me == 'lagarto' and (self.opponent == 'spock' or self.opponent == 'papel'):  
            return 'Vitoria'

        elif self.me == 'tesoura' and (self.opponent == 'lagarto' or self.opponent == 'papel'):  
            return 'Vitoria'
        elif self.me == self.opponent:
            return 'Empate'
        else:
            return 'Derrota'
        